pkg/client/client_v2.go

```diff
@@ -568,6 +578,15 @@ func (c *clientV2) syncUnits(expected *proto.CheckinExpected) {
 		c.unitsStateChanged()
 	}
 
+	var apmConfig *proto.APMConfig
+	func() {
+		c.componentMu.RLock()
+		defer c.componentMu.RUnlock()
+		if c.componentConfig != nil {
+			apmConfig = c.componentConfig.ApmConfig
+		}
+	}()
```

Comment:
Why use an IIFE here? Is it because of the `defer`?

Reply:
it is to guarantee that whatever happens we unlock, yes